xds/internal/xdsclient/client_refcounted_test.go

```diff
@@ -93,26 +93,26 @@ func (s) TestClientNewSingleton(t *testing.T) {
 
 			sCtx, sCancel := context.WithTimeout(ctx, defaultTestShortTimeout)
 			defer sCancel()
-			if _, err := singletonCloseCh.Receive(sCtx); err == nil {
+			if _, err := clientImplCloseCh.Receive(sCtx); err == nil {
 				t.Fatal("singleton client implementation closed before all references are released")
 			}
 		}()
 	}
 
 	// Call the last Close(). The underlying implementation should be closed.
 	closeFunc()
-	if _, err := singletonCloseCh.Receive(ctx); err != nil {
+	if _, err := clientImplCloseCh.Receive(ctx); err != nil {
 		t.Fatalf("Timeout waiting for singleton client implementation to be closed: %v", err)
 	}
 
 	// Calling New() again, after the previous Client was actually closed, should
 	// create a new one.
-	_, closeFunc, err = New()
+	_, closeFunc, err = New("singleton") // TODO(easwars): fix this
 	if err != nil {
 		t.Fatalf("failed to create client: %v", err)
 	}
 	defer closeFunc()
-	if _, err := singletonCreationCh.Receive(ctx); err != nil {
+	if _, err := clientImplCreateCh.Receive(ctx); err != nil {
 		t.Fatalf("Timeout when waiting for singleton xDS client to be created: %v", err)
 	}
 }
```

Comment:
Now that this is no longer a singleton, should we have some tests for the new dimension of numerous clients per name and ref counting/closing across clients of that dimension?

Reply:
Added a test. Thank you.